### PR TITLE
feat: Add goToOverride support to Popup, Search, Bookmarks

### PR DESCRIPTION
**Pull request by kellyhutchins**
- Add goToOverride support to disable animation for bookmarks, search, popup when prefersReducedMotion is enabled. -- Moved bookmarks logic to web map to reduce amount of code in sample.
- Updated focus sample to add search/popup focus logic. I can move this to a separate sample if you think that makes more sense?

### CODE REVIEW

File: demos/consistent-focus.html [hunk 1]
```diff
@@ -239,5 +247,31 @@
...
+
+    // When the popup is visible set focus on it.
+    await reactiveUtils.whenOnce(() => view.popup.visible, signal);
+    view.popup.focus();
+
+    // And when the popup is closed move the focus back to the search wiget.
+    await reactiveUtils.whenOnce(() => !view.popup.visible, signal);
+    searchEl.focusSearch();
+
+  }
```
Comment (geospatialem) [inline demos/consistent-focus.html:273]: 😍

File: demos/consistent-focus.html [hunk 2]
```diff
@@ -18,212 +19,219 @@
...
-  <calcite-panel>
-    <arcgis-map item-id="c2a3444863f2466aaad9efa6e65063e1" id="mapEl">
-      <arcgis-legend legend-style="classic" position="bottom-right"></arcgis-legend>
-    </arcgis-map>
-  </calcite-panel>
+    <calcite-panel>
+      <arcgis-map item-id="c2a3444863f2466aaad9efa6e65063e1" id="mapEl">
+        <arcgis-legend position="bottom-right"></arcgis-legend>
+        <arcgis-expand mode="floating" position="top-right">
+          <arcgis-search></arcgis-search>
```
Comment (geospatialem) [inline demos/consistent-focus.html:104]: With the addition of the search widget, the map UI is looking a bit bloated. WDYT if I moved the `arcgis-legend` to an action/panel combo?

File: demos/consistent-focus.html [hunk 3]
```diff
@@ -18,212 +19,219 @@
...
+      <calcite-action-pad layout="horizontal" expand-disabled slot="content-end">
+        <calcite-action id="toggle-mode" text="Toggle mode" icon="brightness"></calcite-action>
+      </calcite-action-pad>
+      <calcite-tooltip placement="bottom" reference-element="toggle-mode" slot="content-end">Toggle
+        mode</calcite-tooltip>
+    </calcite-navigation>
+
+    <calcite-shell-panel slot="panel-start" display-mode="float">
+      <calcite-action-bar slot="action-bar">
+        <calcite-action data-action-id="layer-effects" icon="effects" text="Layer effects"></calcite-action>
```
Comment (geospatialem) [inline demos/consistent-focus.html:58]: For UI cleanup, WDYT if we move the `arcgis-legend` into an `action` + `panel` combo?

```suggestion
        <calcite-action data-action-id="legend" icon="legend" text="Legend"></calcite-action>
        <calcite-action data-action-id="layer-effects" icon="effects" text="Layer effects"></calcite-action>
```

File: demos/consistent-focus.html [hunk 4]
```diff
@@ -18,212 +19,219 @@
...
+      </calcite-action-bar>
+      <!-- Layer effects -->
```
Comment (geospatialem) [inline demos/consistent-focus.html:60]: For UI cleanup, WDYT if we move the `arcgis-legend` into an `action` + `panel` combo?

```suggestion
      <!-- Legend -->
      <calcite-panel heading="Legend" height-scale="l" data-panel-id="legend" closable>
        <arcgis-legend reference-element="#mapEl"></arcgis-legend>
      </calcite-panel>
      
      <!-- Layer effects -->
```

File: demos/consistent-focus.html [hunk 5]
```diff
@@ -18,212 +19,219 @@
...
+    </calcite-shell-panel>
 
-  <calcite-panel>
-    <arcgis-map item-id="c2a3444863f2466aaad9efa6e65063e1" id="mapEl">
-      <arcgis-legend legend-style="classic" position="bottom-right"></arcgis-legend>
-    </arcgis-map>
-  </calcite-panel>
+    <calcite-panel>
+      <arcgis-map item-id="c2a3444863f2466aaad9efa6e65063e1" id="mapEl">
+        <arcgis-legend position="bottom-right"></arcgis-legend>
```
Comment (geospatialem) [inline demos/consistent-focus.html:102]: For UI cleanup, WDYT if we move the `arcgis-legend` into an `action` + `panel` combo?

```suggestion

```

File: demos/reduced-motion.html [hunk 6]
```diff
@@ -272,140 +262,78 @@
...
-            view.goTo(worstAirResult);
-          });
-        });
+    function overrideGoToOptions(view, goToParams) {
+
+      const { target, options } = goToParams;
+
+      return view.goTo(target, {
+        ...options,
+        animate: !isReduced(),
```
Comment (geospatialem) [inline demos/reduced-motion.html:333]: Beautiful! 🥇

**Review by geospatialem — APPROVED** [review]
Awesome!

Wondering for the consistent focus example if we should modify the UI a bit with the legend? Left a few comments for context, but more of a nitpick. We can chat more next week on this, too.

File: demos/consistent-focus.html (hunk 2 above)
Comment (kellyhutchins) [reply demos/consistent-focus.html:104]: Yep I think moving it to either an expand or action panel would be great.

**Pull request merged**